os/using-environment-variables-in-systemd-units.md

```diff
@@ -24,40 +24,59 @@ Then run `sudo systemctl daemon-reload` and `sudo systemctl restart etcd2.servic
 
 EnvironmentFile similar to Environment directive but reads the environment variables from a text file. The text file should contain new-line-separated variable assignments.
 
-It is impossible to use scripts in Environment directive. So you can not dynamically define Environment, i.e. this doesn't work `Environment=/usr/bin/curl http://example.com/something`. When you need to update your environment values dynamically you can combine systemd service unit and EnvironmentFile directive.
+For example in Container Linux, `flanneld.service` unit file creates `/run/flannel_docker_opts.env` environment file which is used by `docker.service` unit to configure Docker use flannel interface. Here's what the environment file looks like:
 
-For example in Container Linux, `flanneld.service` unit file creates `/run/flannel_docker_opts.env` environment file which is used by `docker.service` unit to configure Docker use flannel interface. You can use similar example in your containers' unit files:
+```sh
+$ cat fleet_machines.service
+DOCKER_OPT_BIP="--bip=10.2.25.1/24"
+DOCKER_OPT_IPMASQ="--ip-masq=false"
+DOCKER_OPT_MTU="--mtu=8951"
+```
+
+The docker unit references this file:
+
+```yaml
+[Unit]
+Description=Docker Application Container Engine
+Documentation=http://docs.docker.com
+After=containerd.service docker.socket network.target
+Requires=containerd.service docker.socket
 
+[Service]
+Type=notify
+EnvironmentFile=-/run/flannel/flannel_docker_opts.env
+<snip>
 ```
-cat fleet_machines.service
+
+Since you can't populate an Environment directive with a script, dynamically setting an environment variable isn't possible.
+
+To accomplish this, you can use the EnvironmentFile directive. Here's a really simple example that downloads some config as environment variables, and then passes it to a container:
```

Comment:
Updated, the other stuff. I dont know enough about the metadata thing to add that.